### DISCUSSION TITLE
Using res.redirect with next v9.5.0

### DISCUSSION THREAD

**Discussion by simonuvarov** [vercel/next.js#15558] [Help] [answered]
Hi,

I've updated next to the latest version (9.5.0) and one of the new features was that now it support `res.redirect` helper function similar to Express. However, when I tried to use it during SSR, it gave me an error:
```
TypeError: res.redirect is not a function
```

I took examples from [this commit](https://github.com/vercel/next.js/pull/14705/commits/540c2295e324a01f7f37f6cb239e589f74502584). Even though those changes were undone later, I suppose that now, when this feature is released, it should work.

My code: 
```javascript
export const getServerSideProps = async context => {
  const { req, res } = context;
  const session = await getCurrentSession({ req });

  if (session) {
    res.redirect('/dashboard');
    return;
  }
  return { props: {} };
};
```

What do I do wrong?

Thanks,
Simon

**Comment by timneutkens** [comment, marked as answer]
These helpers are only available in API routes

**Reply by simonuvarov** [reply]
Okay, thank you for the reply. I got confused because the commit I mentioned used them in the `getServerSideProps` function.

**Reply by timneutkens** [reply]
Looks like that's a bug actually, those example updates should be reverted.

**Reply by simonuvarov** [reply]
They are. That was a pre-revert commit 🙂

**Reply by nicosh** [reply]
With next 9.4 i have this error inside api routes as well.
